pkg/sidecar-controller/snapshot_controller.go

```diff
@@ -331,14 +331,45 @@ func (ctrl *csiSnapshotSideCarController) createSnapshotOperation(content *crdv1
 		return nil, fmt.Errorf("failed to get input parameters to create snapshot for content %s: %q", content.Name, err)
 	}
 
+	// NOTE(xyang): handle create timeout
+	// Add annotation and set to Yes to indicate create snapshot request is
+	// sent to the storage system and wait for a response of success or failure.
+	// Annotation will be set to No only after storage system has responded
+	// with success or failure. If the request times out, retry will happen
+	// and annotation will stay as Yes to avoid leaking of snapshot
+	// resources on the storage system
+	err = ctrl.setAnnVolumeSnapshotBeingCreated(content, utils.AnnVolumeSnapshotBeingCreated_Yes)
+	if err != nil {
+		return nil, fmt.Errorf("failed to set VolumeSnapshotBeingCreated annotation to Yes on the content %s: %q", content.Name, err)
+	}
+
 	driverName, snapshotID, creationTime, size, readyToUse, err := ctrl.handler.CreateSnapshot(content, class.Parameters, snapshotterCredentials)
 	if err != nil {
+		// NOTE(xyang): handle create timeout
+		// If it is not a timeout error, set annotation to No to indicate
+		// storage system has responded with an error
+		errStr := fmt.Sprintf("%q", err)
+		if !strings.Contains(errStr, "DeadlineExceeded") {
```

Comment:
Can we get error codes/types instead of strings?

Reply:
Done